### PR TITLE
Source map improvements

### PR DESCRIPTION
**Pull request by novemberborn**
Bumps `source-map-support`. Now automatically retrieves the source map from the file itself.

I've improved the source map tests. See the commit for all the details, but tl;dr there's now a script to generate one of the fixtures and the other fixtures come from a separate module. Hopefully the tests are easier to understand now. I had a hard time myself!

### CODE REVIEW

File: test/fixture/source-map-file.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const fixture = require('source-map-fixtures').mapFile('throws').require()
```
Comment (sindresorhus) [inline test/fixture/source-map-file.js]: semicolon

File: test/fixture/source-map-file.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+const test = require('../../');
+
+// The uncaught exception is passed to the corresponding cli test. The line
+// numbers from the 'throws' fixture (which uses a map file), as well as the
+// line of the fixture.run() call, should match the source lines.
+test('throw an uncaught exception', t => {
+	setImmediate(run);
+});
+
+const run = () => fixture.run()
```
Comment (sindresorhus) [inline test/fixture/source-map-file.js]: semicolon

File: test/fixture/source-map-inline.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const fixture = require('source-map-fixtures').inline('throws').require()
```
Comment (sindresorhus) [inline test/fixture/source-map-inline.js]: semicolon

File: test/fixture/source-map-inline.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+const test = require('../../');
+
+// The uncaught exception is passed to the corresponding cli test. The line
+// numbers from the 'throws' fixture (using an inline source map), as well as
+// the line of the fixture.run() call, should match the source lines.
+test('throw an uncaught exception', t => {
+	setImmediate(run);
+});
+
+const run = () => fixture.run()
```
Comment (sindresorhus) [inline test/fixture/source-map-inline.js]: semicolon

**Comment by sindresorhus** [comment]
Thanks for working on this @novemberborn. Generally looks good to me.

**Comment by novemberborn** [comment]
@sindresorhus thanks! Semi-colons added.

Any idea why `xo` didn't flag this? I ran `npm test` before submitting…

**Comment by sindresorhus** [comment]
@novemberborn XO ignores fixtures by default.

**Comment by sindresorhus** [comment]
@jamestalmage @vdemedes Looks good to you?

**Comment by vdemedes** [comment]
Yep!

**Comment by jamestalmage** [comment]
:shipit:

**Pull request merged**